helix-term/src/commands/lsp.rs

```diff
@@ -1324,3 +1331,191 @@ pub fn select_references_to_symbol_under_cursor(cx: &mut Context) {
+        Some(
+            lsp::OneOf::Left(true)
+            | lsp::OneOf::Right(lsp::InlayHintServerCapabilities::Options(_)),
+        ) => {
+            let doc_text = &doc.text();
+            let len_lines = doc_text.len_lines();
+
+            // Compute ~3 times the current view height of inlay hints, that way some scrolling
+            // will not show half the view with hints and half without while still being faster
+            // than computing all the hints for the full file (which could be dozens of time
+            // longer than the view is).
+            let view_height = view.inner_height();
+            let first_visible_line = view.estimate_last_doc_line(doc).saturating_sub(view_height);
+            let first_line = first_visible_line
+                .saturating_sub(view_height)
+                .min(len_lines);
+            let last_line = first_visible_line
+                .saturating_add(view_height)
+                .min(len_lines);
+
+            let new_inlay_hint_request = LastInlayHintRequestId {
+                doc_id,
+                doc_revision,
+                first_line,
+                last_line,
+            };
+            // Don't recompute the annotations in case nothing has changed about the view
+            if view.last_inlay_hint_request_id == Some(new_inlay_hint_request) {
+                return None;
+            }
+
+            let doc_slice = doc_text.slice(..);
+            let first_char_in_range = doc_slice.line_to_char(first_line);
+            let last_char_in_range = doc_slice.line_to_char(last_line);
+
+            let range = helix_lsp::util::range_to_lsp_range(
+                doc_text,
+                helix_core::Range::new(first_char_in_range, last_char_in_range),
+                language_server.offset_encoding(),
+            );
+
+            (
+                language_server.text_document_range_inlay_hints(doc.identifier(), range, None),
+                new_inlay_hint_request,
+            )
+        }
+        _ => return None,
+    };
+
+    let callback = super::make_job_callback(
+        future?,
+        move |editor, _compositor, response: Option<Vec<lsp::InlayHint>>| {
+            if !editor.config().lsp.display_inlay_hints {
+                return;
+            }
+
+            // Add annotations to relevant document, not the current one (it may have changed
+            // in between)
+            let doc = match editor.documents.get_mut(&doc_id) {
+                Some(doc) => doc,
+                None => return,
+            };
+            let view = match editor.tree.try_get_mut(view_id) {
+                Some(view) => view,
+                None => {
+                    // If the view doesn't exist anymore, purge its data from the text annotations
+                    // in the document
+                    doc.reset_text_annotations(view_id);
+                    return;
+                }
```

Comment:
this should be handled in 

`Document::remove_view`.

`Document::ensure_view_init` might also be relevant to this PR

Reply:
Done